src/lang-promql/complete/hybrid.ts

```diff
@@ -0,0 +1,231 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 The Prometheus Authors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import { AutocompleteContext, Completion, CompletionResult } from "@codemirror/next/autocomplete";
+import { Complete } from "./index";
+import { PrometheusClient } from "./prometheus/client";
+import { Subtree } from "lezer-tree"
+import { EditorState } from "@codemirror/next/basic-setup";
+
+// TODO: filter every list returned by the prometheusClient
+
+const autocompleteNode = {
+  "MatchOp": [
+    "=",
+    "!=",
+    "=~",
+    "!~"
+  ],
+  "BinOp": [
```

Comment:
ah yeah I forgot it. My bad

Reply:
mmm actually now I remember why I didn't include them. That's because they are not considered are a BinOp currently. And so it's another kind of node to autocomplete them.

Maybe I'm wrong but instead of doing this: 

```
  Expr !and ckw<"and">    BinModifier Expr |
  Expr !and ckw<"unless"> BinModifier Expr |
  Expr !or  ckw<"or">     BinModifier Expr
```

Isn't possible to do this:

```
  Expr !and BinOp<"and">    BinModifier Expr |
  Expr !and BinOp<"unless"> BinModifier Expr |
  Expr !or  BinOp<"or">     BinModifier Expr
```
?

Reply:
Ah right. I think it's a question of whether we want to highlight them differently in lezer/CM. I think I wanted to highlight word-style operators like keywords, and thus I had to make them keywords. So I'm inclined to keep it that way in lezer, but really, they are binary operators that happen to be conditional keywords as well...

Reply:
it's fine for me. It was just just an idea